### PR TITLE
Add specializations of PreconditionChebyshev functions

### PR DESCRIPTION
**Pull request by peterrum**
- specialized version for deal.II vector so that one has to less vector r/w 
- for the case that the matrix does not have `vmult` with pre/post but the preconditioner has

### CODE REVIEW

File: include/deal.II/lac/precondition.h [hunk 1]
```diff
@@ -2492,6 +2497,162 @@ namespace internal
...
+          DEAL_II_OPENMP_SIMD_PRAGMA
+          for (unsigned int i = 0; i < solution_old.locally_owned_size(); ++i)
+            solution_old_ptr[i] = factor1_plus_1 * solution_ptr[i] -
+                                  factor1 * solution_old_ptr[i] +
+                                  temp_vector2_ptr[i] * factor2;
+        }
+
+      solution.swap(solution_old);
+    }
+    template <typename Number,
```
Comment (kronbichler) [inline include/deal.II/lac/precondition.h:2571]:
```suggestion

    template <typename Number,
```

**Review by kronbichler — APPROVED** [review]

**Comment by kronbichler** [comment]
There are some build errors you need to address.

**Comment by peterrum** [comment]
> There are some build errors you need to address.

Fixed!

File: include/deal.II/lac/precondition.h [hunk 2]
```diff
@@ -1194,7 +1199,7 @@ namespace internal
             tmp,
             dst,
             [&](const unsigned int start_range, const unsigned int end_range) {
-              // zero 'tmp' before running the vmult
+              // zero 'temp_vector1' before running the vmult
```
[the remark this answers is not in the document]
Comment (peterrum) [reply include/deal.II/lac/precondition.h]: This was a mistake. `temp_vector1` is used in the Chebyshev context.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**